examples/tcp-retransmit.bpf.c

```diff
@@ -0,0 +1,88 @@
+#include <vmlinux.h>
+#include <bpf/bpf_tracing.h>
+#include "maps.bpf.h"
+#include <bpf/bpf_core_read.h>
+
+#define MAX_ENTRIES 8192
+#define RETRANSMIT  1
+#define TLP         2
+
+#define AF_INET		2
+#define AF_INET6	10
+
+struct ipv4_key_t {
+	u32 saddr;
+	u32 daddr;
+	__u16 sport;
+	__be16 dport;
+    int type;
+};
+
+struct ipv6_key_t {
+	u8 saddr[16];
+	u8 daddr[16];
+	__be16 sport;
+	__u16 dport;
+    int type;
+};
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, struct ipv4_key_t);
+	__type(value, u64);
+} tcp_retransmit_ipv4_count SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, struct ipv6_key_t);
+	__type(value, u64);
+} tcp_retransmit_ipv6_count SEC(".maps");
+
+static int trace_event(void *ctx, const struct sock *skp, int type)
+{
+	__u32 family;
+	family = BPF_CORE_READ(skp, __sk_common.skc_family);
+
+	if (family == AF_INET) {
+		struct ipv4_key_t key;
+
+		BPF_CORE_READ_INTO(&key.saddr, skp, __sk_common.skc_rcv_saddr);
+		BPF_CORE_READ_INTO(&key.daddr, skp, __sk_common.skc_daddr);
+		
+        __portpair portpair = BPF_CORE_READ(skp, __sk_common.skc_portpair);
+        key.sport = (__u16)(portpair & 0xFFFF);
```

Comment:
Reading `.skc_num` and `.skc_dport` seems a bit easier to follow.

Reply:
Changed.